nx/lib/nx.ex

```diff
@@ -4280,6 +4280,116 @@ defmodule Nx do
     )
   end
 
+  @doc """
+  Performs a scatter-add operation on the `target` tensor,
+  adding the `updates` into the corresponding `indices` positions.
+
+  This operation is the grad for `gather/2` and gather-like operations such as
+  `take/3` and `take_along_axis/3`.
+
+  `indices` must be a fully qualified tensor of shape `{n, Nx.rank(target)}`, with `n`
+  being an arbitrary number of indices, while `updates` must have a compatible `{n}` shape.
```

Comment:
What happens for negative indexes? Are we raising for the binary backend?

Reply:
I'm not checking for them!

We can either raise or transform them into the corresponding positive index as we go.

EXLA will probably just break

Reply:
Actually, since I've copied the offset calculation from @jonatanklosko's gather implementation, it actually raises on negative indices :)